test/OffsetHelper.test.ts

```diff
@@ -29,18 +30,23 @@ function parseUSDC(s: string): BigNumber {
 }
 
 describe("OffsetHelper", function () {
-  let offsetHelper: OffsetHelper;
-  let swapper: Swapper;
-  let bct: IToucanPoolToken;
   let nct: IToucanPoolToken;
-  let weth: IERC20;
-  let wmatic: IWETH;
-  let usdc: IERC20;
-  let addr1: SignerWithAddress;
-  let addr2: SignerWithAddress;
-  let addrs: SignerWithAddress[];
-
-  beforeEach(async function () {
+  let bct: IToucanPoolToken;
+  let TOKEN_POOLS = [
+    { name: "NCT", token: () => nct },
+    { name: "BCT", token: () => bct },
+  ];
```

Comment:
I suggest to also define and return `TOKEN_POOLS` in `deployOffsetHelperFixture`.
Then you can remove the undesired `let` declarations of  `bct` and `nct` here and set the `token` field directly, instead of to a getter lambda.

Reply:
That's what I tried to do but since we're using `TOKEN_POOLS` to loop through `it` blocks, I'm unable to extract `TOKEN_POOLS` from the fixture before having to use them. You might be providing a solution in the last sentence but I'm not sure how setting the `token` field directly would look like exactly. 

Would appreciate if you could explain it further!

Reply:
> we're using TOKEN_POOLS to loop through it blocks, I'm unable to extract TOKEN_POOLS from the fixture before having to use them.

Yeah, this makes it unclear to me too

Reply:
Ah I see, sorry, we use the `TOKEN_POOL` outside the `it`. In that case, what we could do is the following.
The fixture doesn't return `nct` and `bct` directly but instead returns `tokens: {nct: ...; bct: ...}`.
Then we simply redefine `const TOKEN_POOLS = ["nct", "bct"]`, that is, as the keys of the `tokens` object. And then we just iterate over the `TOKEN_POOLS` like `for name of TOKEN_POOLS { const poolToken = tokens[name]; ...}`.

Reply:
That makes sense to me. WDTY @SimasG ?

Reply:
I think `TOKEN_POOLS` shouldn't be part of the fixture even if it was possible to put it there. It's just a constant array of two constant strings after all, so it's not "generated" as part of anything happening inside the fixture function. I'd just put it near the top of the file before the outermost `describe`.

Reply:
That's a very fair point!

Reply:
Just committed the changes.

Not very happy with the @ts-ignore lines but wasn't able to find a cleaner solution for typescript's `expression of type 'string' can't be used to index type '{}'.` errors. Happy to implement any suggestions if you have them.

Reply:
Hmm you could try setting the type to a [TS Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) like so `TOKEN_POOLS: Record<string, string> = ...`. Then maybe ts is happy indexing it with a string key.

Reply:
That does it, thank you!